src/ProgramTest.elm

```diff
@@ -880,15 +886,35 @@ clickButton buttonText programTest =
               )
             , ( "a <form> with onSubmit containing an <input type=submit value=" ++ escapeString buttonText ++ "> (not disabled)"
               , simulateHelper functionDescription
-                    (Query.find
-                        [ Selector.tag "form"
-                        , Selector.containing
-                            -- TODO: don't match disabled buttons
-                            [ Selector.tag "input"
-                            , Selector.attribute (Html.Attributes.type_ "submit")
-                            , Selector.attribute (Html.Attributes.value buttonText)
+                    (findButNot
+                        { good =
+                            [ Selector.tag "form"
+                            , Selector.containing
+                                [ Selector.tag "input"
+                                , Selector.attribute (Html.Attributes.type_ "submit")
+                                , Selector.attribute (Html.Attributes.value buttonText)
+                                ]
                             ]
-                        ]
+                        , bads =
+                            [ [ Selector.tag "form"
+                              , Selector.containing
+                                    [ Selector.tag "input"
+                                    , Selector.attribute (Html.Attributes.type_ "submit")
+                                    , Selector.attribute (Html.Attributes.disabled True)
+                                    , Selector.attribute (Html.Attributes.value buttonText)
+                                    ]
+                              ]
+                            ]
+                        , onError =
+                            [ Selector.tag "form"
+                            , Selector.containing
+                                [ Selector.tag "input"
+                                , Selector.attribute (Html.Attributes.type_ "submit")
+                                , Selector.attribute (Html.Attributes.disabled False)
```

Comment:
Nice job guarding against disabled buttons! Too bad there isn't a way to include this check in the base/good query.